### PR TITLE
Add --disable-openmp configure option

### LINKED ISSUE #2131: Add --disable-openmp configure option
We currently "enable" OpenMP (i.e. set the define `LIBMESH_HAVE_OPENMP` and pass the corresponding flags) any time the compiler supports one of the possible OpenMP flags. Basically in threads.m4, we call
```
AX_OPENMP([],[enableopenmp=no])
```
no matter what without any option to disable it. When using helgrind for debugging, it is apparently necessary to have pthreads support turned on but *not* be using OpenMP (@lindsayad made a thread about this on libmesh-devel). So, we need to be able to control these two things independently.

There might also be an even more subtle middle ground where one would want to pass the `-fopenmp` flags to the compiler, but not actually use any OpenMP pragmas in the code... that should also be possible to support if necessary, but I think for now the ability to simply turn off OpenMP is all that's needed.

### PR DESCRIPTION
**Pull request by lindsayad**
Closes #2131

### CODE REVIEW

**Comment by lindsayad** [comment]
@permcody This is to be able to use helgrind meaningfully

File: m4/threads.m4 [hunk 1]
```diff
@@ -32,39 +32,47 @@ AC_DEFUN([ACX_BEST_THREAD],
...
-  dnl If the user explicitly requested the openmp threading model and
-  dnl this compiler doesn't support OpenMP for some reason, we throw an
-  dnl error instead of silently continuing.
-  AS_IF([test "x$requested_thread_model" = "xopenmp"],
-        [
-          AS_IF([test "x$enableopenmp" = "xno"],
-                [AC_MSG_ERROR([requested openmp threading model, but compiler does not support openmp.])])
-        ])
+  AS_IF([test "x$enableopenmp" = "xyes"],
+        [AX_OPENMP([],[enableopenmp=no])]
```
Comment (jwpeterson) [inline m4/threads.m4]: For instance, here there is a closing `]` but no comma, so I guess it doesn't start the third (else) clause of the `AS_IF` statement?

Comment (lindsayad) [reply m4/threads.m4]: Correct. If `enableopenmp = yes`, then we execute the `AX_OPENMP` macro and then we also execute the following macro (the `AS_IF` on whether the macro successfully set `OPENMP_CXXFLAGS`)

Comment (lindsayad) [reply m4/threads.m4]: The corresponding `else` for this `AS_IF` (`test "x$enableopenmp" = "xyes"`) is the `Error if the user simultaneously requested --with-thread-model=openmp and --disable-openmp` bit

**Pull request merged**